pkg/thanos/statefulset.go

```diff
@@ -303,13 +316,37 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 			watchedDirectories = append(watchedDirectories, mountPath)
 		}
 
+		var fields monitoringv1.WebConfigFileFields
+		if tr.Spec.Web != nil {
+			fields = tr.Spec.Web.WebConfigFileFields
+		}
+
+		webConfig, err := webconfig.New(webConfigDir, webConfigSecretName(tr.Name), fields)
+		if err != nil {
+			return nil, err
+		}
+
+		confArg, configVol, configMount, err := webConfig.GetMountParameters()
+		if err != nil {
+			return nil, err
+		}
+		containerArgs = append(containerArgs, fmt.Sprintf("--%s=%s", "http.config", confArg.Value))
+		trVolumes = append(trVolumes, configVol...)
+		trVolumeMounts = append(trVolumeMounts, configMount...)
+
+		if tr.Spec.Web != nil {
+			configReloaderWebConfigFile = confArg.Value
+			configReloaderVolumeMounts = append(configReloaderVolumeMounts, configMount...)
+		}
+
 		additionalContainers = append(
 			additionalContainers,
 			operator.CreateConfigReloader(
 				"config-reloader",
 				operator.ReloaderConfig(config.ReloaderConfig),
+				operator.WebConfigFile(configReloaderWebConfigFile),
 				operator.ReloaderURL(url.URL{
-					Scheme: "http",
+					Scheme: "http", // fixme: should we allow for https?
```

Comment:
Yes the operator should check if the web config enables HTTPS (see what's been done for Alertmanager).